pkg/cli/cmds/server.go

```diff
@@ -143,6 +143,11 @@ func ServerRun(clx *cli.Context) error {
 
 func validateCNI(clx *cli.Context) {
 	cni := clx.String("cni")
+	if strings.HasSuffix(cni, "+multus") {
```

Comment:
`multus,primary`

Reply:
brad to open an issue for helm-controller ordering bit for future work

Reply:
Ordering issue: https://github.com/rancher/rke2/issues/960

@manuelbuil let me know if you need pointers on converting this over to a StringSliceFlag. StringSliceFlags have the unfortunate downside of not clearing the default value from the list if you provide values on the CLI, so you have to hack around it not providing a default value in the flag, adding the default text to the usage string, and manually setting the default if the list is empty.

Probably looking at something like:
```go
cnis := []string{}
for _, cni := range clx.StringSlice("cni") {
  for _, v := range strings.Split(",") {
    cnis = append(cnis, v)
  }
}

switch len(cnis) {
case 0:
  cnis = append(cnis, "canal")
  clx.Set("disable", "rke2-multus")
case 1:
  if cnis[0] == "multus" {
    logrus.Fatal("invalid value provided for --cni flag: multus must be used alongside another primary cni selection")
  }
case 2:
  if cnis[0] == "multus" {
    cnis = cnis[:1]
  } else {
    logrus.Fatal("invalid values provided for --cni flag: may only provide multiple values if multus is the first value")
  }
default:
  logrus.Fatal("invalid values provided for --cni flag: may not provide more than two values")
}  
```
and then switch the bit down below to operating on `cnis[0]` instead of `cni`

Reply:
Thanks!